src/Root.cc

```diff
@@ -41,20 +42,32 @@ class sdf::RootPrivate
   /// \brief The worlds specified under the root SDF element
   public: std::vector<World> worlds;
 
+  /// \brief A model, light or actor under the root SDF element
+  /// This variant does not currently retain ownership, and instead points to
+  /// the first model/light/actor if they exist. When the vectors below are
+  /// removed this should be changed from a variant of pointers to a variant of
+  /// objects
+  public: std::variant<std::monostate, sdf::Model*, sdf::Light*, sdf::Actor*>
+              modelLightOrActor;
+
   /// \brief The models specified under the root SDF element
-  public: std::vector<Model> models;
+  /// Deprecated: to be removed in libsdformat12
+  public: std::vector<sdf::Model> models SDF_DEPRECATED(11.0);
```

Comment:
I'm not sure if deprecating these member variables is useful here, as compiling this file creates a bunch of warnings.

Reply:
Yeah, I would only deprecate the public API. Stuff in `RootPrivate` is all private.